lib/AST/ActorIsolation.cpp

```diff
@@ -183,6 +191,9 @@ bool ActorIsolation::isEqual(const ActorIsolation &lhs,
     auto *lhsActor = lhs.getActorInstance();
     auto *rhsActor = rhs.getActorInstance();
     if (lhsActor && rhsActor) {
+      if (lhsActor == rhsActor)
+        return true;
+
```

Comment:
Could the FIXME right below this be addressed now that we're tracking captures?

Reply:
I guess this condition isn't so bad; it's trying to make same-name captures of `self` be considered the same isolation as the original `self`. So I think the answer is no, I can't fix that FIXME to work for arbitrary captures without additional changes to the AST, because we only track this for `self`.

The condition that terrifies me is the one where we don't have `VarDecl`s.